operators/multiclusterobservability/controllers/multiclusterobservability/multiclusterobservability_controller.go

```diff
@@ -378,6 +394,12 @@ func (r *MultiClusterObservabilityReconciler) SetupWithManager(mgr ctrl.Manager)
 
 	secretPred := predicate.Funcs{
 		CreateFunc: func(e event.CreateEvent) bool {
+			if e.Object.GetNamespace() == config.GetDefaultNamespace() &&
+				e.Object.GetName() == config.ThanosObjectStoreSecretName {
+				commonutil.AddBackupLabelToSecret(c, config.ThanosObjectStoreSecretName, config.GetDefaultNamespace())
+				return true
```

Comment:
we should not return true here, since create thanos-object-storage secret will not lead to any changes on the rendered resources.
same as the UpdateFunc.

If you want to add logic to watch the secret and make it applied automatically, need to delete the pods which mount the secret to forcely trigger a pod restart

Reply:
Fixed in latest commit